src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/IHandlerProvider.cs

```diff
@@ -16,5 +14,5 @@ public interface IHandlerProvider
 {
     ImmutableArray<RequestHandlerMetadata> GetRegisteredMethods();
 
-    IMethodHandler GetMethodHandler(string method, Type? requestType, Type? responseType);
+    ImmutableArray<Lazy<IMethodHandler, string?>> GetMethodHandlers(string method, Type? requestType, Type? responseType);
```

Comment:
what's the string in this lazy?

Reply:
For the metadata

Reply:
what is the metadata i nthis case?  Generally, we use an interface in these cases.  like `Lazy<IConfigurationFixProvider, CodeChangeProviderMetadata>` so that the metadata is strongly typed.

Reply:
We have LanguageMetadata and ILanguageMetadata for this purpose.

Reply:
Reverted back to single handler with optional language param